core/js/src/PlatformSpecifics.kt

```diff
@@ -4,13 +4,17 @@
  */
 package kotlinx.datetime.internal
 
-import kotlinx.datetime.internal.JSJoda.ZoneId
+import kotlinx.datetime.internal.JSJoda.ZoneRulesProvider
 
-internal actual fun getAvailableZoneIdsSet(): Set<String> =
-    ZoneId.getAvailableZoneIds().unsafeCast<Array<String>>().toSet()
+internal actual fun readTzdb(): Pair<List<String>, List<String>>? = try {
+    val tzdbData = ZoneRulesProvider.asDynamic().getTzdbData()
```

Comment:
I did manage to find an incantation that seems like it would allow us to improve the lives of JS and Wasm/JS kotlinx-datetime users, but that's too difficult to be in the scope of this PR.

Reply:
Thanks for the clarification!